### PR TITLE
[ZkTracer] Add several `blake` test cases

### LINKED ISSUE #3116: [ZkTracer] Add more `blake` test cases
🔐 🔐 `Should this be a security advisory instead?` 🔐 🔐

## Description
Write a clear and concise description of what you want to happen. Add any considered drawbacks.

## Motivation
This section should indicate what value we are receiving from it.

## Tasks
- [ ] Medium-level task that needs to be implemented for this feature request to be considered complete.

## Acceptance criteria
- [ ] List tests that need to be implemented or checks to consider this task completed. 

## Risks
- [ ] These are risks that may prevent completion of the task as well as areas to pay special attention to

## Remember to
 - [ ] Add the `documentation` label in case there is an impact on the documentation
 - [ ] Add `priority` and `team` labels
 - [ ] Add Task for updating the Runbook or adding/updating existing metrics and alerts.

### PR DESCRIPTION
**Pull request by lorenzogentile404**
This PR implements issue(s) #

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] If this change is deployed to any environment (including Devnet), E2E test coverage exists or is included in this
  PR.
* [ ] I have informed the team of any breaking changes if there are any.

---

> [!NOTE]
> **Low Risk**
> Low risk: changes are confined to the RISC-V example/test harness (Makefile/docs) and add/commit test fixtures, with no impact on production runtime code.
> 
> **Overview**
> Adds a new `make blake-all` target in `arithmetization/src/test/examples/Makefile` to iterate over input vectors from `rust/src/blake/blake.all` and run `blake_with_in_bytes.rs` for each case.
> 
> Updates the examples `README.md` accordingly (including the shell alias rename to `riscv-test`) and commits a large set of generated single-opcode RV64IM assembly test programs under `asm/src/op/` by removing the previous ignore rule; the old opcode-test generation/runner scripts under `asm/scripts/` are deleted.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d0a782c158ad789e15f3fc56f4a8431947c015da. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: arithmetization/src/test/examples/rust/src/blake/run_blake2f_all.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+cd "$(dirname "$0")/.."
```
Comment (amkCha) [inline arithmetization/src/test/examples/rust/src/blake/run_blake2f_all.sh]: It's better to have a Make command to run all test cases than a python script - so we only have Make commands as a format everywhere across our stack/ the monorepo

File: arithmetization/src/test/examples/rust/src/blake/Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		selector == "" { print NR ":" $$0; next } \
+		selector ~ /^[0-9]+$$/ && NR == selector { print NR ":" $$0; next } \
+		selector ~ /^[0-9]+-[0-9]+$$/ && NR >= bounds[1] && NR <= bounds[2] { print NR ":" $$0 } \
+	' "$(VECTORS)") || exit 1; \
+	[[ -n "$$cases" ]] || { echo "no cases matched: $$selector"; exit 1; }; \
+	failures=""; \
+	while IFS=: read -r n in_bytes; do \
+		printf 'case %s... ' "$$n"; \
+		output=$$(make --no-print-directory -f "$(EXAMPLES_MAKEFILE)" TEST="$(RUST_TEST)" IN_BYTES="$$in_bytes" 2>&1) || true; \
+		result_line=$$(printf '%s\n' "$$output" | sed '/^[[:space:]]*$$/d' | tail -n 2 | head -n 1); \
```
[the remark this answers is not in the document]
Comment (lorenzogentile404) [reply arithmetization/src/test/examples/rust/src/blake/Makefile]: There is another line afterwards printing a separator, so this works fine.

File: arithmetization/src/test/examples/asm/src/op/op_add.s [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.section .text
+.global _start
```
Comment (amkCha) [inline arithmetization/src/test/examples/asm/src/op/op_add.s:2]: why are you adding these files here ?

File: arithmetization/src/test/examples/rust/src/blake/Makefile [hunk 4]
```diff
@@ -0,0 +1,43 @@
+SHELL := /bin/bash
+
+THIS_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+EXAMPLES_MAKEFILE := $(abspath $(THIS_DIR)/../../..)/Makefile
+VECTORS := $(THIS_DIR)blake2f.all
+RUST_TEST := blake/blake_with_in_bytes.rs
+SELECTOR ?=
+
+.PHONY: all
```
Comment (amkCha) [inline arithmetization/src/test/examples/rust/src/blake/Makefile]: we are going to have too many Makefiles if we have one per folder - can you put the command lines at the root of the test folder ? Will have 1 Makefile for the arith folder and one for the test folder more specifically

File: arithmetization/src/test/examples/asm/src/op/op_add.s (hunk 3 above)
Comment (lorenzogentile404) [reply arithmetization/src/test/examples/asm/src/op/op_add.s:2]: before there was a python script generating them. Now I just added the generated small tests and removed the python script.

File: arithmetization/src/test/examples/rust/src/blake/Makefile (hunk 4 above)
Comment (lorenzogentile404) [reply arithmetization/src/test/examples/rust/src/blake/Makefile]: this `Makefile` is specific for `blake`, while `arithmetization/src/test/examples/Makefile` is to run a generic test. I believe it would be confusing to have them at in the same folder or merge them. What do you think?

File: arithmetization/src/test/examples/asm/src/op/op_add.s (hunk 3 above)
Comment (amkCha) [reply arithmetization/src/test/examples/asm/src/op/op_add.s:2]: ok those are the ones that were AI written, not the test ACT4 ones right ?

Comment (lorenzogentile404) [reply arithmetization/src/test/examples/asm/src/op/op_add.s:2]: they are just small simple tests to easily see what happens in zkc, but they can also be removed as the ACT4 tests already cover corner cases extensively for sure.

File: arithmetization/src/test/examples/rust/src/blake/Makefile (hunk 4 above)
Comment (amkCha) [reply arithmetization/src/test/examples/rust/src/blake/Makefile]: I think it's ok as long as the command line naming mentions it :+1: In the end you know, we might have a command line for generic tests indeed, and one for all rust tests

File: arithmetization/src/test/examples/asm/src/op/op_add.s (hunk 3 above)
Comment (lorenzogentile404) [reply arithmetization/src/test/examples/asm/src/op/op_add.s:2]: exactly, they were written by Claude. ACT4 are way bigger.

File: arithmetization/src/test/examples/rust/src/blake/Makefile [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+all:
+	@set -euo pipefail; \
```
Comment (amkCha) [inline arithmetization/src/test/examples/rust/src/blake/Makefile]: I believe this bit could be trimmed as it's doing and printing a lot of unnecessary things maybe - so it could benefit from 
1) renaming the function as we were saying
2) either pass, and we don't do anything, or fail, and make the command line fail (not just a "fail" print), while printing the selectors that failed

**Review by amkCha — APPROVED** [review]
